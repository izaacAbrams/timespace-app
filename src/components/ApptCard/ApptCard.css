.Appointments__notes {
  font-size: 16px;
}
