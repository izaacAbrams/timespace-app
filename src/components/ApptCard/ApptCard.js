import React, { Component } from "react";
import moment from "moment";

export default class ApptCard extends Component {
  render() {
    const apptTime = moment(this.props.appt).format("LT");
    return (
      <div className="Appointments__main_appt">
        <h3>{this.props.name}</h3>
        <p>Service: {this.props.service}</p>
        <p>Time: {apptTime}</p>
        <button>Edit</button>
        <button>Delete</button>
      </div>
    );
  }
}
