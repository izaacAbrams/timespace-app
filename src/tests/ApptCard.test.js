import React from "react";
import ReactDOM from "react-dom";
import ApptCard from "../components/ApptCard/ApptCard.js";
import renderer from "react-test-renderer";

describe("apptcard component", () => {
  it("renders test appointment", () => {
    const div = document.createElement("div");
    const appt = {
      name: "Test Appt",
      appt_date_time: new Date().toISOString(),
      service: "Test",
    };
    ReactDOM.render(<ApptCard appt={appt} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders the UI as expected", () => {
    const appt = {
      name: "Test Appt",
      appt_date_time: new Date("2029-01-22T16:28:32.615Z").toISOString(),
      service: "Test",
    };
    const tree = renderer.create(<ApptCard appt={appt} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
