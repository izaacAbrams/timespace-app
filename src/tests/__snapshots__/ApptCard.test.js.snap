// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apptcard component renders the UI as expected 1`] = `
<div
  className="Appointments__main_appt"
>
  <h3>
    Test Appt
  </h3>
  <p>
    Service: 
    Test
  </p>
  <p>
    Time: 
    10:28 AM
  </p>
  <button
    onClick={[Function]}
  >
    Edit
  </button>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
